cdap-docs/reference-manual/source/glossary.rst

```diff
@@ -63,6 +63,12 @@ Glossary
       synchronous calls into CDAP from an external system and perform server-side
       processing on-demand. They are similar to a stored procedure in a traditional database
       system. Procedures are typically used to post-process data at query time.
+
+   Worker
+      Workers are typically long-running background programs that can be used to execute tasks.
+      Each instance of a worker runs either in its own YARN container (CDAP distributed mode) or
+      a single thread (CDAP Standalone or In-Memory mode) and the number of instances may be updated
+      via RESTful APIs or the CLI. Datasets can be accessed from inside Workers.
```

Comment:
Datasets can be accessed. Services can be discovered but that is true for Flows, MapReduce etc. We can't access any other program inside Worker.